src/pkce.ts

```diff
@@ -0,0 +1,24 @@
+import crypto from "crypto";
+
+function base64URLEncode(buffer: Buffer) {
+  return buffer
+    .toString("base64")
+    .replace(/\+/g, "-")
+    .replace(/\//g, "_")
+    .replace(/=/g, "");
+}
```

Comment:
Node.js 14 supports `buffer.toString('base64url')` directly, so this util function is not needed.